src/foraging_gui/Foraging.py

```diff
@@ -1216,6 +1223,105 @@ def _GetSettings(self):
         self.Other_lick_spout_distance=self.Settings['lick_spout_distance_box'+str(self.box_number)]
         self.rig_name = '{}'.format(self.current_box)
 
+    def _ConnectSlims(self):
+        '''
+            Connect to Slims
+        '''
+        try:
+            self.slims_client = SlimsClient(username=os.environ['SLIMS_USERNAME'],
+                                            password=os.environ['SLIMS_PASSWORD'])
+        except KeyError:
+            raise KeyError('SLIMS_USERNAME and SLIMS_PASSWORD do not exist as '
+                         'environment variables on machine. Please add')
+
+        try:
+            self.slims_client.fetch_model(models.SlimsMouseContent, barcode='00000000')
```

Comment:
It looks like the logic here is you try to check SLIMS for mouse "0000000". If you get an exception saying "no record found" then you assume that means you connected to slims. If you get some other error, then you assume its a connection issue. 

Does the SLIMS API have a better way to check the connection? 

If fetch fails for other reasons (like the SLIMS server is down). Then the error message is a little confusing. I would say something like. "Exception trying to read from Slims: {e}.\n This may be due to credential errors. ..."

Reply:
Yeah that makes sense I'll ask Patrick if there is a better way to tell if the credentials are incorrect